docs/F5-vs-HAProxy.md

```diff
@@ -0,0 +1,43 @@
+# F5 or HAProxy Decision
+
+This document contains the information used to decide whether to leverage the built-in OpenShift HAProxy router as the default cluster ingress for HTTP(S) or the F5 BigIP Controller operator.
+
+## Background
+
+The F5 BigIP Controller operator was originally identified as the preferred tool to enable non-HTTP ingress to the OCP4 clusters.  The default OpenShift HAProxy routers do not support this type of traffic, and the integration with the F5 controller will enable automation and consistent implementation of these (non-HTTP) ingress types.
```

Comment:
CRD support is currently ALPHA and does not yet support non-http traffic. https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/docs/_static/config_examples/crd/CustomResource.md#how-cis-works-with-crds

We will need to build our own operator to mediate between a custom object in the namespace (say a VirtualServerClaim) and the CIS CRDs (VirtualServer) so that we can manage handing out of IP:PORT pairs and not have teams overriding each other.

ConfigMap support can handle non-http, but is not workable in a multi-tenant environment. Could be used for testing while we wait for the CRD support to mature.

The BigIP can do Web Application Firewallsing, Bot Defense, and DDOS protection. Some teams may wish to leverage those features. Thought that could likely be added later on.

CIS support for OpenShift Routes is also available in a limited mode where only routes with a specific label are picked up instead of all routes.

Reply:
Is there something that needs to be changed in the background to highlight those details as part of a decision for the default Router to be Internal OCP or External F5?

Reply:
Nah, I think this was mostly me brain dumping for you.